### PR TITLE
[GOBBLIN-1983] Remove Optionals to make DagManager, EventSubmitter, and TopologyCatalog required for GaaS operation

### PR DESCRIPTION
**Pull request by arjun4084346**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1983

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
fields `isInstrumentationEnabled` needs to be always enabled and `eventSubmitter` should always be present for GaaS to function. Removing `Optional` from them.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
trivial changes

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java [hunk 1]
```diff
@@ -255,6 +259,10 @@ public static void markMeter(Optional<Meter> meter) {
     markMeter(meter, 1);
   }
 
+  public static void markMeter(Meter meter) {
```
Comment (umustafi) [inline gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java:262]: where is this being used?

Comment (phet) [reply gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java:262]: my related Q is whether we still require the `Optional` version or could *replace* them w/ this one

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java [hunk 2]
```diff
@@ -284,7 +264,7 @@ public void orchestrate(Spec spec, Properties jobProps, long triggerTimestampMil
           return;
         }
         Dag<JobExecutionPlan> compiledDag = compiledDagOptional.get();
-        if (compiledDag == null || compiledDag.isEmpty()) {
+        if (compiledDag.isEmpty()) {
           FlowCompilationValidationHelper.populateFlowCompilationFailedEventMessage(eventSubmitter, spec, flowMetadata);
           Instrumented.markMeter(this.flowOrchestrationFailedMeter);
```
Comment (arjun4084346) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java:269]: It is used here, the parameter changed from Optional<X> to X

**Review by umustafi — APPROVED** [review]

File: gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java [hunk 3]
```diff
@@ -41,6 +39,8 @@
 import com.google.common.collect.Lists;
 import com.google.common.io.Closer;
 
+import javax.annotation.Nonnull;
```
Comment (phet) [inline gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java:42]: when I see:
> Import order: java, org, com, gobblin.
here - https://gobblin.apache.org/docs/developer-guide/CodingStyle/

I had always presume `javax` would go next to `java`, not between `com` and `gobblin`.  are you sure this is supposed to move here?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java [hunk 4]
```diff
@@ -109,10 +109,10 @@ public class Orchestrator implements SpecCatalogListener, Instrumentable {
 
   private final ClassAliasResolver<SpecCompiler> aliasResolver;
 
+  @Inject
   public Orchestrator(Config config, Optional<TopologyCatalog> topologyCatalog, Optional<DagManager> dagManager,
```
Comment (phet) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java:113]: I'd argue that, similarly to the `EventSubmitter` the `DagManager` is no longer justifiably `Optional`.  while we're at it, can we clean this up too?

also, what of the `Optional<TopologyCatalog>`?

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java [hunk 5]
```diff
@@ -66,7 +66,7 @@ public final class FlowCompilationValidationHelper {
    * caller.
    * @param flowSpec
    * @param optionalFlowExecutionId for scheduled (non-ad-hoc) flows, to pass the ID "laundered" via the DB;
-   *                                see: {@link MysqlMultiActiveLeaseArbiter javadoc section titled
+   *                                see: {@link org.apache.gobblin.runtime.api.MysqlMultiActiveLeaseArbiter javadoc section titled
```
Comment (phet) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java:69]: why was this necessary?  (I didn't see you remove an import)

**Review by phet — COMMENTED** [review]
nice cleanup, arjun!

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java (hunk 5 above)
Comment (arjun4084346) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java:69]: just to fix javadoc

File: gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java (hunk 3 above)
Comment (arjun4084346) [reply gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java:42]: I do not do any import changes manually, intelliJ does it, hopefully based on the provided codestyle

File: gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java (hunk 1 above)
Comment (arjun4084346) [reply gobblin-core-base/src/main/java/org/apache/gobblin/instrumented/Instrumented.java:262]: There are many non-gaas callers to this method with Optional parameter, which may be providing parameter that is truly optional.

**Review by phet — APPROVED** [review]
whew, what a relief to pare down all this out-of-date and out-of-use code - very nice!
